java/client/src/test/java/glide/connectors/resources/ThreadPoolResourceAllocatorTest.java

```diff
@@ -0,0 +1,43 @@
+package glide.connectors.resources;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+import io.netty.channel.EventLoopGroup;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.Test;
+
+public class ThreadPoolResourceAllocatorTest {
+
+    ThreadPoolResourceAllocator service;
+
+    @Test
+    public void getOrCreateReturnsDefault() {
```

Comment:
what is this test protecting against? 
it's mocks all the way through.

Reply:
That the default ELG is cached and returned each time.

Reply:
I don't see this contract documented anywhere. Is this a contract that requires enforcement, or is it just an artifact of the current implementation?

Reply:
It should be contractual I will add documentation for it in getOrCreate.